org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/IDecompiler.java

```diff
@@ -31,4 +31,57 @@ public interface IDecompiler extends IContentProvider {
 	 * @throws CoreException
 	 */
 	public String getSource(IClassFile classFile, IProgressMonitor monitor) throws CoreException;
+
+	/**
+	 * Provide decompiled source code from a resource.
+	 *
+	 * @param classFile
+	 *            the class file to decompile
+	 * @param monitor
+	 *            monitor of the activity progress
+	 * @return text content or <code>null</code>
+	 * @throws CoreException
+	 */
+	default public DecompilerResult getDecompiledSource(IClassFile classFile, IProgressMonitor monitor) throws CoreException {
```

Comment:
nit: We can remove the `public` modifier, according to [JLS](https://docs.oracle.com/javase/specs/jls/se7/html/jls-9.html#jls-9.4):

`It is permitted, but discouraged as a matter of style, to redundantly specify the public and/or abstract modifier for a method declared in an interface.`